frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/status-card/utils.ts

```diff
@@ -31,14 +31,14 @@ export const getCephHealthState: PrometheusHealthHandler = (responses = [], erro
 };
 
 export const getDataResiliencyState: PrometheusHealthHandler = (responses = [], errors = []) => {
-  if (errors[0]) {
+  const progress: string = getResiliencyProgress(responses[0]);
+  if (errors[0] || !progress) {
```

Comment:
@bipuladh `!'0'` would return `false` but `!0` would not.

Resiliency need to be shown for case 0 not `Unknown` status

Reply:
```suggestion
  if (errors[0] || Number.isFinite(progress)) {
```
Lets try and remove redundant type conversions.

Reply:
Unfortunately, this does not removes redundancy.
We are still going through this chain of `number->string->number->string` conversion.
This will just move the last conversion one level up (in resiliency function).